docs/unit-testing.qmd

```diff
@@ -0,0 +1,120 @@
+---
+title: Unit testing
+editor:
+  markdown:
+    wrap: sentence
+---
+
+Testing Shiny apps is important to ensure the shiny app functions as expected and to catch any bugs or errors before deployment. It helps maintain code quality, user experience, and prevents potential issues from reaching the end-users.
```

Comment:
From https://github.com/posit-dev/py-shiny-site/pull/167#discussion_r1680121551 . Re-commented as the original file name changed

```suggestion
Testing is important to ensure your apps continue working as intended. 
It helps maintain code quality, user experience, and prevents potential issues from reaching the end-users.

There are two main approaches to testing Shiny apps: unit testing and end-to-end testing.
Relatively speaking, unit tests are limited compared to end-to-end, since can't actually launch your app in a browser -- they're only useful for testing your servers-side Python logic.

In this article, we'll provide an intro to unit testing with **pytest**. See the next article for end-to-end testing.
```